bindings/wasm/examples/browser/resolve_history.js

```diff
@@ -0,0 +1,130 @@
+    // set to the last DID document published.
+    const diff2 = doc.diff(doc2, diff1Receipt.messageId, key);
+
+    // Publish the diff to the Tangle, starting a diff chain.
+    const diff2Receipt = await client.publishDiff(updatedMessageId, diff2);
+    if (log) logToScreen("Diff Chain Update (2):");
+    if (log) logExplorerUrlToScreen(getExplorerUrl(doc, diff2Receipt.messageId));
+
+    // ===========================================================================
+    // DID History 1
+    // ===========================================================================
+
+    // Retrieve the message history of the DID.
+    const history1 = await client.resolveHistory(doc.id.toString());
+
+    // The history shows one document in the integration chain (plus the current document), and two
+    // diffs in the diff chain.
+    if (log) logToScreen("History (1):")
+    if (log) logObjectToScreen(history1);
+
+    // ===========================================================================
+    // Integration Chain Update 2
+    // ===========================================================================
+
+    // Publish an integration chain update, which writes the full updated DID document to the Tangle.
+    // Note: the previousMessageId points to the messageId of the last integration chain update,
+    //       not the last diff chain message.
+    doc2.previousMessageId = updatedMessageId;
+
+    // Sign the DID Document with the appropriate key.
+    doc2.sign(key);
+
+    // Publish the Identity to the IOTA Network, this may take a few seconds to complete Proof-of-Work.
+    const intChainUpdateReceipt = await client.publishDocument(doc.toJSON());
+
+    // Log the results.
+    if (log) logToScreen("Int. Chain Update (2):")
+    if (log) logExplorerUrlToScreen(getExplorerUrl(doc, intChainUpdateReceipt.messageId));
+
+    // ===========================================================================
+    // DID History 2
+    // ===========================================================================
+
+    // Retrieve the updated message history of the DID.
+    const history2 = await client.resolveHistory(doc.id.toString());
+
+    // The history now shows two documents in the integration chain (plus the current document), and no
+    // diffs in the diff chain. This is because the previous document published included those updates
+    // and we have not added any diffs pointing to the latest document.
+    if (log) logToScreen("History (2):")
+    if (log) logObjectToScreen(history2);
+
+    // ===========================================================================
+    // Diff Chain History
+    // ===========================================================================
+
+    // Fetch the diff chain of the previous integration chain message.
+    // We can still retrieve old diff chains, but they do not affect DID resolution.
+    let verificationMethod = doc.authentication();
```

Comment:
`resolve_history.js:126 Uncaught (in promise) TypeError: doc.authentication is not a function
    at resolveHistory (resolve_history.js:126)`

Reply:
Did you rebuild the bindings with `npm run build`? The `authentication` method was added in one of my later commits.